### PR TITLE
Remove dependency on sklearn for regression example.

### PR DESCRIPTION
**Pull request by dellis23**
And numpy.

### CODE REVIEW

File: examples/regression.py [hunk 1]
```diff
@@ -3,18 +3,25 @@
...
 import torch
 import torch_mlir
 
 
 # Dataset
-X, y, coef = make_regression(n_features=3, coef=True)
-X = torch.from_numpy(X).to(dtype=torch.float32)
-y = torch.from_numpy(y).to(dtype=torch.float32)
-X, X_test, y, y_test = train_test_split(X, y)
+X = torch.tensor((
```
Comment (silvasean) [inline examples/regression.py:11]: It should be possible to generate this with a handful of lines of code. You can generate a holdout target weight/bias matrix (not seen by the training procedure), and then run random inputs through it and add some noise.

File: examples/regression.py [hunk 2]
```diff
@@ -3,18 +3,25 @@
...
-from sklearn.model_selection import train_test_split
 import torch
 import torch_mlir
 
 
 # Dataset
-X, y, coef = make_regression(n_features=3, coef=True)
-X = torch.from_numpy(X).to(dtype=torch.float32)
-y = torch.from_numpy(y).to(dtype=torch.float32)
-X, X_test, y, y_test = train_test_split(X, y)
```
Comment (silvasean) [inline examples/regression.py:17]: train_test_split can be done with a one-liner indexing. Something like
split = int(0.9 * num_examples)
X, X_test = X[:split], X[split:]

**Review by silvasean — CHANGES_REQUESTED** [review]

**Comment by silvasean** [comment]
Also in the example in the IREE repo it would be good to generate a larger amount of random data on the fly there -- there isn't really that much signal that the loss is going down with a tiny number of examples.

File: examples/regression.py (hunk 1 above)
Comment (dellis23) [reply examples/regression.py]: Done.

File: examples/regression.py (hunk 2 above)
Comment (dellis23) [reply examples/regression.py:17]: Yeah, I removed it altogether because I thought it was unnecessary for illustrating the example and ends up just adding visual noise to the code.  I've added it back and done it manually.

File: examples/regression.py [hunk 3]
```diff
@@ -8,20 +8,17 @@
...
-    -32.02257999,
-    36.08020568,
-    -177.70424428,
-    -22.68094929,
-), dtype=torch.float32)
+def make_training_data():
+    coefficients = torch.tensor([3.0, 4.0, 5.0], dtype=torch.float32)
+    bias = torch.tensor(6.0, dtype=torch.float32)
+    X, y = [], []
+    for i in range(1000):
```
Comment (silvasean) [inline examples/regression.py]: I think it should be possible to do this without the Python loop. E.g. torch.rand(1000, 3), ...

**Review by silvasean — APPROVED** [review]

File: examples/regression.py (hunk 3 above)
Comment (dellis23) [reply examples/regression.py]: Thanks, much simpler.  I should keep an eye out for those opportunities.
